tests/Core.nut

```diff
@@ -0,0 +1,422 @@
+// 1) Apply default values to 'params' if undefined
+// 2) Create Rocky instance
+// 3) Set Rocky's handlers from 'params' if defined (Rocky.authorize, Rocky.onUnauthorized, Rocky.onTimeout, Rocky.onNotFound, Rocky.onException)
+// 4) Set Rocky's middlewares from 'params' if defined (Rocky.use)
+// 5) Create handler for incoming requests (Rocky.Route)
+// 6) Set Rocky.Route's handlers from 'params' if defined (Rocky.Route.authorize, Rocky.Route.onUnauthorized, Rocky.Route.onTimeout, Rocky.Route.onException)
+// 7) Set Rocky.Route's middlewares from 'params' if defined (Rocky.Route.use)
+// 8) Send one (or more) request(s) and wait for Rocky's response.
+//    When get response, verify statuscode and call callback function for additional verify (if defined). If everything ok, then test passed, otherwise failed.
+//    If more than one requests send, then test will be passed only when all responses verifications passed.
+// 
+// @param {table} params
+// @param {string} expect - Expected completion of the test (success|fail)
+// @return {Promise}
+function createTest(params = {}, expect = "success") {
+    return Promise(function(resolve, reject) {
+        local success = function() {
+            if (expect == "fail") {
+                reject("createTest test was resolved, but 'fail' expected");
+            } else {
+                resolve();
+            }
+        };
+        local fail = function(reason = null) {
+            if (expect == "fail") {
+                resolve();
+            } else {
+                reject(reason);
+            }
+        };
+        local app;
+        local route;
+        // Setup default values
+        params.setdelegate(defaultParams);
+        try {
+            // Call Rocky.Route constructor directly
+            if (params.paramsRockyRoute != null) {
+                local illegalRoute = Rocky.Route(params.paramsRockyRoute);
+                if (params.onAuthorizeRoute != null) {
+                    illegalRoute.authorize(params.onAuthorizeRoute);
+                }
+                if (params.onUnauthorizedRoute != null) {
+                    illegalRoute.onUnauthorized(params.onUnauthorizedRoute);
+                }
+                if (params.onTimeoutRoute != null) {
+                    illegalRoute.onTimeout(params.onTimeoutRoute);
+                }
+                if (params.onExceptionRoute != null) {
+                    illegalRoute.onException(params.onExceptionRoute);
+                }
+                illegalRoute.use(params.mw);
+            }
+            // Call Rocky.Context constructor directly
+            if (params.paramsRockyContext != null) {
+                local illegalContext = Rocky.Context(params.paramsRockyContext[0], params.paramsRockyContext[1]);
+                if (params.paramsRockyContextAdditionalUsage != null && params.paramsRockyContextAdditionalUsage) {
+                    local tmp;
+                    tmp = illegalContext.isComplete();

```

Comment:
It simply creates a Rocky.Context with invalid parameters, and then verifies that calling methods and reading from fields of the instance of Rocky.Context will (or not) result in an error. Used in negative tests that call Rocky.Context constructor directly.